File: system-test/dotnet/Driver.Port/Shop/Dtos/ViewOrderResponse.cs
namespace Driver.Port.Shop.Dtos;

public class ViewOrderResponse
{
    public required string OrderNumber { get; set; }
    public required DateTime OrderTimestamp { get; set; }
    public required string Sku { get; set; }
    public required int Quantity { get; set; }
    public required decimal UnitPrice { get; set; }
    public required decimal TotalPrice { get; set; }
    public required OrderStatus Status { get; set; }
}

File: system-test/java/src/main/java/com/optivem/shop/dsl/driver/port/shop/dtos/ViewOrderResponse.java
package com.optivem.shop.dsl.driver.port.shop.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ViewOrderResponse {
    private String orderNumber;
    private Instant orderTimestamp;
    private String sku;
    private int quantity;
    private BigDecimal unitPrice;
    private BigDecimal totalPrice;
    private OrderStatus status;
}
